apps/src/applab/ProtectedDesignWorkspace.jsx

```diff
@@ -0,0 +1,44 @@
+import { ApplabInterfaceMode } from './constants';
+import React from 'react';
+import { connect } from 'react-redux';
+
+const ProtectedDesignWorkspace = React.createClass({
+  propTypes: {
+    interfaceMode: React.PropTypes.oneOf([
+      ApplabInterfaceMode.CODE,
+      ApplabInterfaceMode.DESIGN,
+      ApplabInterfaceMode.DATA
+    ]).isRequired
+  },
+
+  shouldComponentUpdate: function (nextProps) {
+    // This component is currently semi-protected. We don't want to completely
```

Comment:
In our ProtectedStatefulDiv "Protected" means we completely disallow rerendering - the component will be rendered once, and we'll warn if we ever try to render it again.

Protected here means we're okay with rerendering, but we control which props we allow to change such that we can be reasonably sure the DOM structure won't change.

Maybe these two concepts are similar enough that we don't need different names for them, but when I originally read the component name, I was surprised to find that it was willing to rerender itself.